src/test/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManagerTest.java

```diff
@@ -0,0 +1,89 @@
+package uk.ac.cam.cl.dtg.segue.api.managers;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static uk.ac.cam.cl.dtg.segue.api.managers.UserAccountManager.isEmailValid;
+import static uk.ac.cam.cl.dtg.segue.api.managers.UserAccountManager.isUserNameValid;
+
+public class UserAccountManagerTest {
+
+    @ParameterizedTest
+    @MethodSource("namesToTest")
+    public void nameValidation(boolean expectedIsValid, String name) {
+        boolean actualIsValid = isUserNameValid(name);
+        assertEquals(expectedIsValid, actualIsValid);
+    }
+
+    private static Stream<Arguments> namesToTest() {
+        return Stream.of(
+                Arguments.of(true, "testname"), // Lowercase is valid
+                Arguments.of(true, "TESTNAME"), // Uppercase is valid
+                Arguments.of(true, "TestName"), // Mixture of cases is valid
+                Arguments.of(true, "TestName123"), // Numbers are valid
+                Arguments.of(true, "Test_Name"), // Underscores are valid
+                Arguments.of(true, "Test Name"), // Double-barrelled names with spaces are valid
+                Arguments.of(true, "Test-Name"), // Double-barrelled names with hyphens are valid
+                Arguments.of(true, "O'Name"), // Apostraphes are valid
+                Arguments.of(false, null), // Null is not valid
+                Arguments.of(false, ""), // Empty string is not valid
+                // Names exceeding the maximum length of 255 characters are not valid
+                Arguments.of(false, "TestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestNameTestName"),
```

Comment:
I think it would be more clear and less prone to errors to replace this string with something like `"x".repeat(256);`